bpf/tests/ipsec_redirect_native.c

```diff
@@ -76,3 +65,141 @@ int ipsec_redirect_check(__maybe_unused struct __ctx_buff *ctx)
 
 	test_finish();
 }
+
+static __always_inline
+int bad_identities_check(__maybe_unused struct __ctx_buff *ctx, bool ipv4)
+{
+	test_init();
+
+	int ret = 0;
+	__be16 proto = ipv4 ? bpf_htons(ETH_P_IP) : bpf_htons(ETH_P_IPV6);
+
+	/* fill encrypt map with node's current SPI 3 */
+	struct encrypt_config cfg = {
+		.encrypt_key = BAD_SPI,
+	};
+	map_update_elem(&cilium_encrypt_state, &ret, &cfg, BPF_ANY);
+
+	if (ipv4)
+		ipcache_v4_add_entry(SOURCE_IP, 0,
+				     SOURCE_IDENTITY, 0, BAD_SPI);
+	else
+		ipcache_v6_add_entry((const union v6addr *)SOURCE_IP_6, 0,
+				     SOURCE_IDENTITY, 0, BAD_SPI);
+
```

Comment:
These would only become relevant if we pass `UNKNOWN_ID` to `ipsec_maybe_redirect_to_encrypt()`, correct?